pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -1352,25 +1361,268 @@ func (vs *VSphere) NodeExists(c *govmomi.Client, nodeName k8stypes.NodeName) (bo
 	return false, nil
 }
 
-// Creates a folder using the specified name.
-// If the intermediate level folders do not exist,
-// and the parameter createParents is true,
-// all the non-existent folders are created.
-func makeDirectoryInDatastore(c *govmomi.Client, dc *object.Datacenter, path string, createParents bool) error {
+func (vs *VSphere) createDummyVMWithDiskAttached(ctx context.Context, datacenter *object.Datacenter, datastore *object.Datastore, volumeOptions *VolumeOptions) (string, error) {
+	var diskFormat string
+
+	// Default diskformat as 'thin'
+	if volumeOptions.DiskFormat == "" {
+		volumeOptions.DiskFormat = ThinDiskType
+	}
+
+	if _, ok := diskFormatValidType[volumeOptions.DiskFormat]; !ok {
+		return "", fmt.Errorf("Cannot create disk. Error diskformat %+q."+
+			" Valid options are %s.", volumeOptions.DiskFormat, DiskformatValidOptions)
+	}
+
+	diskFormat = diskFormatValidType[volumeOptions.DiskFormat]
+
+	// Generate a UUID
+	uuidVal := string(uuidgenerator.NewUUID())
```

Comment:
Do we even allow simultaneous provisioning? I remember Abrar complaining that provisioning requests are serialized. Considering these VMs are short lived, I would discount collisions for better user experience since these requests will show up as tasks in VC UI with long vm name.